src/go/rpk/pkg/cli/transform/list.go

```diff
@@ -75,11 +78,24 @@ The --detailed flag (-d) opts in to printing extra per-processor information.
 			p, err := p.LoadVirtualProfile(fs)
 			out.MaybeDie(err, "unable to load config: %v", err)
 
-			api, err := adminapi.NewClient(fs, p)
-			out.MaybeDie(err, "unable to initialize admin api client: %v", err)
+			var l []adminapi.TransformMetadata
+			if p.FromCloud && !p.CloudCluster.IsServerless() {
```

Comment:
Should we have a special error message for serverless?

Reply:
Good point... Added an error message in `deploy, list, and delete` https://github.com/redpanda-data/redpanda/pull/17023/commits/8187221597016ec3950c5fe72030ed03eab7d022

Thanks!